core/src/calleeFunctions/helpers/oneInch.ts

```diff
@@ -0,0 +1,197 @@
+import { ethers } from 'ethers';
+import { getCalleeAddressByCollateralType } from '../../constants/CALLEES';
+import { getCollateralConfigBySymbol } from '../../constants/COLLATERALS';
+import { getContractSymbolByAddress } from '../../contracts';
+import { getChainIdByNetworkType } from '../../network';
+import { CollateralConfig } from '../../types';
+import BigNumber from '../../bignumber';
+import { getTokenAddressByNetworkAndSymbol } from '../../tokens';
+import { WAD_NUMBER_OF_DIGITS } from '../../constants/UNITS';
+
+const EXPECTED_SIGNATURE = '0x12aa3caf'; // see https://www.4byte.directory/signatures/?bytes4_signature=0x12aa3caf
+
+export const getOneInchUrl = (chainId: number) => {
+    return `https://api.1inch.io/v5.0/${chainId}`;
+};
+
+interface Protocol {
+    id: string;
+    title: string;
+    img: string;
+    img_color: string;
+}
+
+interface OneInchToken {
+    symbol: string;
+    name: string;
+    address: string;
+    decimals: 0;
+    logoURI: string;
+}
+interface OneInchSwapRepsonse {
+    fromToken: OneInchToken;
+    toToken: OneInchToken;
+    toTokenAmount: string;
+    fromTokenAmount: string;
+    protocols: OneInchSwapRoute[];
+    tx: {
+        from: string;
+        to: string;
+        data: string;
+        value: string;
+        gasPrice: string;
+        gas: string;
+    };
+}
+interface LiquiditySourcesResponse {
+    protocols: Protocol[];
+}
```

Comment:
Nit: there is no much sense to type http response if you don't validate received data (except maybe for code completion in the editor)

Reply:
i agree, i would love to have the response typed anyway for convenience